### PR TITLE
Fix file activity dialog

### PR DESCRIPTION
**Pull request by claucambra**
This PR fixes the file activity dialog, which shows up blank when clicking on certain file activities (e.g. deleted file activities). 

Signed-off-by: Claudio Cambra <claudio.cambra@gmail.com>

### CODE REVIEW

File: src/gui/fileactivitylistmodel.cpp [hunk 1]
```diff
@@ -26,26 +26,15 @@ FileActivityListModel::FileActivityListModel(QObject *parent)
     setDisplayActions(false);
 }
 
-void FileActivityListModel::load(AccountState *accountState, const QString &localPath)
+void FileActivityListModel::load(OCC::AccountState *accountState, const int objectId)
```
[the remark this answers is not in the document]
Comment (claucambra) [reply src/gui/fileactivitylistmodel.cpp:29]: I was getting a warning on QtCreator that slots needs fully qualified argument types, but it seems to work without it too. Will revert

**Review by allexzander — APPROVED** [review]
Left one comment, minor. Tested the PR, seems to be working.

File: src/gui/fileactivitylistmodel.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (claucambra) [reply src/gui/fileactivitylistmodel.cpp]: Changed

File: src/gui/tray/activitylistmodel.h [hunk 2]
```diff
@@ -48,6 +48,8 @@ class ActivityListModel : public QAbstractListModel
         ActionIconRole = Qt::UserRole + 1,
         AccountRole,
         ObjectTypeRole,
+        ObjectIdRole,
+        ObjectNameRole,
```
Comment (mgallien) [inline src/gui/tray/activitylistmodel.h:52]: looks like you are replacing `AbsolutePathRole` by `ObjectIdRole` can you remove it ?

File: src/gui/fileactivitylistmodel.cpp [hunk 3]
```diff
@@ -61,10 +50,12 @@ void FileActivityListModel::startFetchJob()
     QObject::connect(job, &JsonApiJob::jsonReceived,
         this, &FileActivityListModel::activitiesReceived);
 
+    qDebug() << "FETCHING FILE WITH ID " << _objectId;
```
Comment (mgallien) [inline src/gui/fileactivitylistmodel.cpp:53]: can you check if this is needed and if you want to keep it, please use categorized logging like we do in most other places

File: src/gui/tray/activitylistmodel.h (hunk 2 above)
Comment (claucambra) [reply src/gui/tray/activitylistmodel.h:52]: AbsolutePathRole is still in the enum

Comment (claucambra) [reply src/gui/tray/activitylistmodel.h:52]: Sorry I misunderstood you, I thought you were asking if it had been removed. I will remove it

Comment (claucambra) [reply src/gui/tray/activitylistmodel.h:52]: Okay, removed

**Comment by mgallien** [comment]
can you have a look at existing automated tests for the model to increase coverage of the new lines of code ?
`ActivityListModel` should be easy
probably best to ignore the other as they looks mostly untested

**Comment by claucambra** [comment]
> can you have a look at existing automated tests for the model to increase coverage of the new lines of code ? `ActivityListModel` should be easy probably best to ignore the other as they looks mostly untested

Added a couple of lines for the new code in the data function of the activitylistmodel

**Review by mgallien — APPROVED** [review]

**Pull request merged**